packages/web/components/assets/pool-assets-icon.tsx

```diff
@@ -14,61 +14,53 @@ export const PoolAssetsIcon: FunctionComponent<Props> = ({
-          "absolute z-10 flex items-center justify-center overflow-hidden"
-        )}
-      >
-        {assets[0].coinImageUrl ? (
-          <Image
-            src={assets[0].coinImageUrl}
-            alt={assets[0].coinDenom}
-            width={size === "md" ? 50 : 40}
-            height={size === "md" ? 50 : 40}
-          />
-        ) : (
-          <Image
-            src="/icons/question-mark.svg"
-            alt="no token icon"
-            width={size === "md" ? 50 : 40}
-            height={size === "md" ? 50 : 40}
-          />
-        )}
-      </div>
-      <div
-        className={classNames(
-          {
-            "ml-10 h-[3.125rem] w-[3.125rem]": size === "md",
-            "ml-5 h-[2.5rem] w-[2.5rem]": size === "sm",
-          },
-          "flex shrink-0 items-center justify-center overflow-hidden"
-        )}
-      >
-        {assets.length >= 3 ? (
-          <div className="body1 ml-2.5 text-white-mid">{`+${
-            assets.length - 1
-          }`}</div>
-        ) : assets[1].coinImageUrl ? (
-          <Image
-            src={assets[1].coinImageUrl}
-            alt={assets[1].coinDenom}
-            width={size === "md" ? 50 : 40}
-            height={size === "md" ? 50 : 40}
-          />
-        ) : (
-          <Image
-            src="/icons/question-mark.svg"
-            alt="no token icon"
-            width={size === "md" ? 50 : 40}
-            height={size === "md" ? 50 : 40}
-          />
-        )}
-      </div>
+    <div
+      style={{ width: `${assets.slice(undefined, 4).length * 30}px` }}
+      className={classNames(
+        "relative flex h-fit items-center",
+        size === "md" ? "h-[50px]" : "h-[40px]"
+      )}
+    >
+      {assets.map(({ coinDenom, coinImageUrl }, index) =>
+        index > 3 ? null : (
+          <div
+            key={coinDenom}
+            style={{
+              marginLeft: size === "md" ? `${index * 28}px` : `${index * 24}px`,
+              zIndex: 50 - index,
+            }}
+            className={classNames(
+              {
+                [`h-[3.125rem] w-[3.125rem]`]: size === "md",
+                [`h-[2.5rem] w-[2.5rem]`]: size === "sm",
+                "shrink-0": index > 0,
+              },
+              "absolute flex items-center justify-center overflow-hidden"
+            )}
+          >
+            {index > 2 ? (
+              <div className="body1 pl-4 text-white-mid">{`+${
+                assets.length - 1
+              }`}</div>
+            ) : coinImageUrl ? (
+              <Image
+                src={coinImageUrl}
+                alt={coinDenom}
+                width={size === "md" ? 50 : 40}
+                height={size === "md" ? 50 : 40}
+              />
+            ) : (
+              <Image
+                src="/icons/question-mark.svg"
+                alt="no token icon"
+                width={size === "md" ? 50 : 40}
+                height={size === "md" ? 50 : 40}
+              />
+            )}
```

Comment:
We could use the `denom-image` component here. Also, we could modify to fallback to use the question mark